TriblerGUI/widgets/channel_torrent_list_item.py

```diff
@@ -161,15 +161,18 @@ def on_health_response(self, response):
         self.update_health(total_seeders, total_leechers)
 
     def update_health(self, seeders, leechers):
-        if seeders > 0:
-            self.health_text.setText("good health (S%d L%d)" % (seeders, leechers))
-            self.set_health_indicator(STATUS_GOOD)
-        elif leechers > 0:
-            self.health_text.setText("unknown health (found peers)")
-            self.set_health_indicator(STATUS_UNKNOWN)
-        else:
-            self.health_text.setText("no peers found")
-            self.set_health_indicator(STATUS_DEAD)
+        try:
+            if seeders > 0:
+                self.health_text.setText("good health (S%d L%d)" % (seeders, leechers))
+                self.set_health_indicator(STATUS_GOOD)
+            elif leechers > 0:
+                self.health_text.setText("unknown health (found peers)")
+                self.set_health_indicator(STATUS_UNKNOWN)
+            else:
+                self.health_text.setText("no peers found")
+                self.set_health_indicator(STATUS_DEAD)
+        except RuntimeError:
+            self._logger.error("The underlying GUI widget has already been removed.")
```

Comment:
This is more of a side note, but at some point we should probably make a wrapper to make sure that the current object has not been destructed. At this rate we'll be adding `try except RuntimeError` everywhere.